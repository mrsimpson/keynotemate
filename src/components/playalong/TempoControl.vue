<template>
  <div class="tempo-control">
    <div class="tempo-input">
      <label for="tempo">{{ t('playback.tempo') }}:</label>
      <input
        id="tempo"
        v-model="localTempo"
        max="240"
        min="30"
        type="number"
        step="1"
        @change="updateTempo"
      />
      <span>{{ t('playback.bpm') }}</span>
    </div>

    <select class="tempo-preset" @change="handlePresetChange">
      <option value="">{{ t('playback.presets.label') }}</option>
      <option value="40">{{ t('playback.presets.slow') }}</option>
      <option value="80">{{ t('playback.presets.medium') }}</option>
      <option value="120">{{ t('playback.presets.fast') }}</option>
    </select>
  </div>
</template>

<script setup lang="ts">
import { ref, watch } from 'vue';
import { useI18n } from 'vue-i18n';

const { t } = useI18n();

const props = defineProps<{
  tempo: number;
  onTempoChange: (tempo: number) => void;
}>();

const localTempo = ref(props.tempo);

watch(
  () => props.tempo,
  (newTempo) => {
    localTempo.value = newTempo;
  },
);

const updateTempo = () => {
  const tempo = Math.max(30, Math.min(240, localTempo.value));
  props.onTempoChange(tempo);
};

const handlePresetChange = (event: Event) => {
  const value = (event.target as HTMLSelectElement).value;
  if (value) {
    localTempo.value = parseInt(value, 10);
    updateTempo();
  }
};
</script>

<style scoped>
.tempo-control {
  display: flex;
  align-items: center;
  gap: var(--spacing-md);
}

.tempo-input {
  display: flex;
  align-items: center;
  gap: var(--spacing-sm);
}

.tempo-input input {
  width: 70px;
  padding: var(--spacing-sm);
  border: 1px solid #ddd;
  border-radius: var(--radius-sm);
  text-align: center;
}

.tempo-preset {
  padding: var(--spacing-sm);
  border: 1px solid #ddd;
  border-radius: var(--radius-sm);
  background: white;
  font-size: 1rem;
  cursor: pointer;
  color: var(--text);
}

@media (max-width: 600px) {
  .tempo-control {
    flex-direction: column;
    align-items: stretch;
    width: 100%;
  }

  .tempo-preset {
    width: 100%;
  }
}
</style>
