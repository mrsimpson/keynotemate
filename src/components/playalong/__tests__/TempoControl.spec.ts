import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import { createI18n } from 'vue-i18n';
import TempoControl from '../TempoControl.vue';

describe('TempoControl Component', () => {
  const i18n = createI18n({
    legacy: false,
    locale: 'en',
    messages: {
      en: {
        playback: {
          tempo: 'Tempo',
          bpm: 'BPM',
          presets: {
            label: 'Preset',
            slow: 'Slow (40 BPM)',
            medium: 'Medium (80 BPM)',
            fast: 'Fast (120 BPM)',
          },
        },
      },
    },
  });

  it('renders tempo control snapshot', () => {
    const wrapper = mount(TempoControl, {
      props: {
        tempo: 60,
        onTempoChange: () => {},
      },
      global: {
        plugins: [i18n],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders tempo control with different tempo snapshot', () => {
    const wrapper = mount(TempoControl, {
      props: {
        tempo: 120,
        onTempoChange: () => {},
      },
      global: {
        plugins: [i18n],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
