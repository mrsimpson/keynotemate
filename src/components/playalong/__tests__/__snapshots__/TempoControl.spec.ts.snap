// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TempoControl Component > renders tempo control snapshot 1`] = `
"<div data-v-b5d43374="" class="tempo-control">
  <div data-v-b5d43374="" class="tempo-input"><label data-v-b5d43374="" for="tempo">Tempo:</label><input data-v-b5d43374="" id="tempo" max="240" min="30" type="number" step="1"><span data-v-b5d43374="">BPM</span></div><select data-v-b5d43374="" class="tempo-preset">
    <option data-v-b5d43374="" value="">Preset</option>
    <option data-v-b5d43374="" value="40">Slow (40 BPM)</option>
    <option data-v-b5d43374="" value="80">Medium (80 BPM)</option>
    <option data-v-b5d43374="" value="120">Fast (120 BPM)</option>
  </select>
</div>"
`;

exports[`TempoControl Component > renders tempo control with different tempo snapshot 1`] = `
"<div data-v-b5d43374="" class="tempo-control">
  <div data-v-b5d43374="" class="tempo-input"><label data-v-b5d43374="" for="tempo">Tempo:</label><input data-v-b5d43374="" id="tempo" max="240" min="30" type="number" step="1"><span data-v-b5d43374="">BPM</span></div><select data-v-b5d43374="" class="tempo-preset">
    <option data-v-b5d43374="" value="">Preset</option>
    <option data-v-b5d43374="" value="40">Slow (40 BPM)</option>
    <option data-v-b5d43374="" value="80">Medium (80 BPM)</option>
    <option data-v-b5d43374="" value="120">Fast (120 BPM)</option>
  </select>
</div>"
`;
